models/product_template.py

```diff
@@ -0,0 +1,29 @@
+from odoo import fields, models, api
+
+
+class ProductTemplate(models.Model):
+    _inherit = 'product.template'
+
+    is_flower = fields.Boolean()
+    flower_id = fields.Many2one('flower', compute='_compute_flower', inverse='_set_flower')
```

Comment:
This is a good working solution; however there is no need to make the field computed nor give it an inverse function. What you did would be great if we were using product variants, but for the sake of this exercise, we are not doing that